### PR TITLE
🟡(change): cache organization response

### PR DESCRIPTION
**Pull request by ruancomelli**
Organization is pulled quite a bit from the application, but should not be updated often.

## Summary by Sourcery

This pull request introduces caching for organization data using Redis, removes the rainbow animation feature from the theme provider, and consolidates organization-related API endpoints and hooks. Additionally, it updates dependencies to include the new Redis client library.

- **New Features**:
    - Introduced caching for organization data using Redis to reduce database load and improve response times.
- **Enhancements**:
    - Removed the rainbow animation feature from the theme provider and related components.
    - Consolidated organization-related API endpoints and hooks for better consistency and maintainability.
- **Chores**:
    - Updated dependencies in go.mod to include new Redis client library.

### CODE REVIEW

**Comment by ruancomelli** [comment]
This is an experiment review for experiment `improve_review_guide`
Run ID: `improve_review_guide/run_2024-08-06T18-06-29_v1-21-0-70-g56484f776-dirty`
The benchmark review for this pull request can be found at https://github.com/sourcery-ai-experiments/Trenova/pull/14.
This pull request was cloned from `https://github.com/sourcery-ai-experiments/Trenova/pull/11`. _(Note: the URL is not a link to avoid triggering a notification on the original pull request.)_

<details>
<summary>Experiment configuration</summary>

```yaml
review_config:
  # User configuration for the review
  # - benchmark - use the user config from the benchmark reviews
  # - <value> - use the value directly
  user_review_config:
    enable_ai_review: true
    enable_rule_comments: false

    enable_complexity_comments: false
    enable_security_comments: false
    enable_tests_comments: false
    enable_comment_suggestions: false
    enable_functionality_review: false

    enable_pull_request_summary: false
    enable_review_guide: true

    enable_approvals: false

  ai_review_config:
    # The model responses to use for the experiment
    # - benchmark - use the model responses from the benchmark reviews
    # - llm - call the language model to generate responses
    model_responses:
      comments_model: benchmark
      comment_area_model: benchmark
      comment_validation_model: benchmark
      comment_suggestion_model: benchmark
      complexity_model: benchmark
      docstrings_model: benchmark
      functionality_model: benchmark
      security_model: benchmark
      tests_model: benchmark
      pull_request_summary_model: benchmark
      review_guide_model: llm

# The pull request dataset to run the experiment on
pull_request_dataset:
  # open-source PRs
- https://github.com/sourcery-ai-experiments/gpt-crawler/pull/1
- https://github.com/sourcery-ai-experiments/graphviz2drawio/pull/1
- https://github.com/sourcery-ai-experiments/graphviz2drawio/pull/2
- https://github.com/sourcery-ai-experiments/iptux/pull/5
- https://github.com/sourcery-ai-experiments/iptux/pull/6
- https://github.com/sourcery-ai-experiments/kalima/pull/3
- https://github.com/sourcery-ai-experiments/PED/pull/3
- https://github.com/sourcery-ai-experiments/pydantic_tfl_api/pull/1
- https://github.com/sourcery-ai-experiments/pydantic_tfl_api/pull/2
- https://github.com/sourcery-ai-experiments/Trenova/pull/11
  # Sourcery PRs
- https://github.com/sourcery-ai/core/pull/4055
- https://github.com/sourcery-ai/core/pull/4263
- https://github.com/sourcery-ai/core/pull/4352
- https://github.com/sourcery-ai/core/pull/4364
- https://github.com/sourcery-ai/core/pull/4366
- https://github.com/sourcery-ai/core/pull/4390
- https://github.com/sourcery-ai/core/pull/4402
- https://github.com/sourcery-ai/core/pull/4445
- https://github.com/sourcery-ai/core/pull/4462
- https://github.com/sourcery-ai/core/pull/4463

review_comment_labels: []

# Benchmark reviews generated by running
#   python -m scripts.experiment benchmark <experiment_name>
benchmark_reviews:
- dataset_pull_request: https://github.com/sourcery-ai-experiments/gpt-crawler/pull/1
  review_pull_request: https://github.com/sourcery-ai-experiments/gpt-crawler/pull/5
- dataset_pull_request: https://github.com/sourcery-ai-experiments/graphviz2drawio/pull/1
  review_pull_request: https://github.com/sourcery-ai-experiments/graphviz2drawio/pull/14
- dataset_pull_request: https://github.com/sourcery-ai-experiments/graphviz2drawio/pull/2
  review_pull_request: https://github.com/sourcery-ai-experiments/graphviz2drawio/pull/13
- dataset_pull_request: https://github.com/sourcery-ai-experiments/iptux/pull/5
  review_pull_request: https://github.com/sourcery-ai-experiments/iptux/pull/22
- dataset_pull_request: https://github.com/sourcery-ai-experiments/iptux/pull/6
  review_pull_request: https://github.com/sourcery-ai-experiments/iptux/pull/23
- dataset_pull_request: https://github.com/sourcery-ai-experiments/kalima/pull/3
  review_pull_request: https://github.com/sourcery-ai-experiments/kalima/pull/11
- dataset_pull_request: https://github.com/sourcery-ai-experiments/PED/pull/3
  review_pull_request: https://github.com/sourcery-ai-experiments/PED/pull/7
- dataset_pull_request: https://github.com/sourcery-ai-experiments/pydantic_tfl_api/pull/1
  review_pull_request: https://github.com/sourcery-ai-experiments/pydantic_tfl_api/pull/8
- dataset_pull_request: https://github.com/sourcery-ai-experiments/pydantic_tfl_api/pull/2
  review_pull_request: https://github.com/sourcery-ai-experiments/pydantic_tfl_api/pull/9
- dataset_pull_request: https://github.com/sourcery-ai-experiments/Trenova/pull/11
  review_pull_request: https://github.com/sourcery-ai-experiments/Trenova/pull/14
- dataset_pull_request: https://github.com/sourcery-ai/core/pull/4055
  review_pull_request: https://github.com/sourcery-ai-experiments/core/pull/207
- dataset_pull_request: https://github.com/sourcery-ai/core/pull/4263
  review_pull_request: https://github.com/sourcery-ai-experiments/core/pull/208
- dataset_pull_request: https://github.com/sourcery-ai/core/pull/4352
  review_pull_request: https://github.com/sourcery-ai-experiments/core/pull/209
- dataset_pull_request: https://github.com/sourcery-ai/core/pull/4364
  review_pull_request: https://github.com/sourcery-ai-experiments/core/pull/210
- dataset_pull_request: https://github.com/sourcery-ai/core/pull/4366
  review_pull_request: https://github.com/sourcery-ai-experiments/core/pull/211
- dataset_pull_request: https://github.com/sourcery-ai/core/pull/4390
  review_pull_request: https://github.com/sourcery-ai-experiments/core/pull/212
- dataset_pull_request: https://github.com/sourcery-ai/core/pull/4402
  review_pull_request: https://github.com/sourcery-ai-experiments/core/pull/213
- dataset_pull_request: https://github.com/sourcery-ai/core/pull/4445
  review_pull_request: https://github.com/sourcery-ai-experiments/core/pull/214
- dataset_pull_request: https://github.com/sourcery-ai/core/pull/4462
  review_pull_request: https://github.com/sourcery-ai-experiments/core/pull/215
- dataset_pull_request: https://github.com/sourcery-ai/core/pull/4463
  review_pull_request: https://github.com/sourcery-ai-experiments/core/pull/216

```

</details>

**Comment by SourceryAI** [comment]
## Reviewer's Guide by Sourcery

This pull request introduces caching for organization data using Redis, removes the rainbow animation feature from the theme provider, and consolidates organization-related API endpoints and hooks. Additionally, it updates dependencies to include the new Redis client library.

### File-Level Changes

| Files | Changes |
| ----- | ------- |
| `internal/api/services/organization.service.go`<br/>`internal/server/server.go`<br/>`pkg/redis/client.go`<br/>`config/server_config.go` | Introduced Redis caching for organization data, including client initialization and cache management methods. |
| `web/frontend/src/components/ui/theme-provider.tsx`<br/>`web/frontend/src/components/layout/user-avatar-menu.tsx`<br/>`web/frontend/src/components/layout/topbar.tsx` | Removed rainbow animation feature from the theme provider and related components. |
| `web/frontend/src/hooks/useQueries.ts`<br/>`web/frontend/src/services/OrganizationRequestService.ts`<br/>`web/frontend/src/components/layout/logo.tsx`<br/>`web/frontend/src/pages/admin/Dashboard.tsx` | Consolidated organization-related API endpoints and hooks for better consistency and maintainability. |

---

<details>
<summary>Tips</summary>

- Trigger a new Sourcery review by commenting `@sourcery-ai review` on the pull request.
- Continue your discussion with Sourcery by replying directly to review comments.
- You can change your review settings at any time by accessing your [dashboard](https://app.sourcery.ai):
    - Enable or disable the Sourcery-generated pull request summary or reviewer's guide;
    - Change the review language;
- You can always [contact us](mailto:support@sourcery.ai) if you have any questions or feedback.

</details>

File: internal/api/services/organization.service.go [hunk 1]
```diff
@@ -40,30 +43,36 @@ func NewOrganizationService(s *server.Server) *OrganizationService {
...
+	return fmt.Sprintf("organization:%s", orgID.String())
+}
 
-	err := s.db.NewSelect().
-		Model(org).
-		Relation("State").
-		Where("organization.business_unit_id = ?", buID).
-		Where("organization.id = ?", orgID).
-		Scan(ctx)
+func (s *OrganizationService) GetOrganization(ctx context.Context, buID, orgID uuid.UUID) (*models.Organization, error) {
```
Comment (SourceryAI) [inline internal/api/services/organization.service.go:55]: **suggestion:** Refactor cache invalidation to reduce code duplication.

The repeated pattern of cache invalidation in multiple methods can be refactored into a separate method to improve maintainability.

**Review by SourceryAI — COMMENTED** [review]
Hey @ruancomelli - I've reviewed your changes - here's some feedback:

**Overall Comments**:
- Consider encapsulating repetitive error-handling patterns for cache operations into utility functions to keep the code cleaner.
- Ensure thorough testing of the refactored `useQueries` hook and API endpoints to avoid breaking existing functionality.
- Fix the typo in the comment: 'Try toi fetch the organization from the cache' should be 'Try to fetch the organization from the cache'.

<details>
<summary>Here's what I looked at during the review</summary>

- 🟡 **General issues**: 1 issue found
- 🟢 **Documentation**: all looks good
</details>

***

[LangSmith trace](https://smith.langchain.com/o/7d8518ad-6dd3-5c7b-96d2-757769296335/projects/p/88560225-5072-48bc-967c-dc54da53f265/r/96c5e078-7ca7-49b2-ae0d-0de77ba27830?poll=true)

<sub>
Help me be more useful! Please click 👍 or 👎 on each comment to tell me if it was helpful.
</sub>

**Pull request closed without merging**